src/Browscap/Data/DataCollection.php

```diff
@@ -425,7 +425,16 @@ public function addSourceFile($src)
                     ) {
                         throw new \UnexpectedValueException(
                             'the key "' . $child['match']
-                            . '" is defined with platform placeholder, but no platforms are asigned'
+                            . '" is defined with platform placeholder, but no platforms are assigned'
+                        );
+                    }
+
+                    if (false !== strpos($child['match'], '#DEVICE#')
```

Comment:
When testing my changes for the YaApp, the Standard tests were failing and I couldn't figure out why.

It ended up being because I had left the `#DEVICE#` placeholder in the "device-less" iOS block, so the patterns going into the file still had `#DEVICE#` in them.

This check here will make sure that if the `#DEVICE#` placeholder is present, that the child has the `devices` object as well.

After adding this, and testing a build, it found a few cases where there was a `#DEVICE#` token, but no `devices`. I've fixed those as well (seen above).